comp/logs/agent/agent.go

```diff
@@ -136,7 +142,12 @@ func (a *agent) setupAgent() error {
 	} else {
 		status.SetCurrentTransport(status.TransportTCP)
 	}
-	inventories.SetAgentMetadata(inventories.AgentLogsTransport, string(status.GetCurrentTransport()))
+	// The severless agent doesn't use FX for now. This means that the logs agent will not have 'inventoryAgent'
+	// initialized for serverless. This is ok since metadata is not enabled for serverless.
+	// TODO: (components) - This condition should be removed once the serverless agent use FX.
+	if a.inventoryAgent != nil {
+		a.inventoryAgent.Set(logsTransport, string(status.GetCurrentTransport()))
```

Comment:
Having `Set` makes sense but I am wondering if you considered using fx group instead?

Reply:
I think, for now, it's a bit overkill since this method is only used in two places and we have no plan to add more data to this payload.